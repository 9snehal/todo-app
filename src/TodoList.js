import React, { Component } from "react";
import { css } from 'react-emotion';
import TodoItems from './TodoItems';
import './TodoList.css';
import Loader from './Loader';
import { RaisedButton } from "material-ui";
import { logout } from './helpers/auth';

const appTokenKey = "appToken"; // also duplicated in Login.js

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
   items: [],
   loading: true
 };
    this.addItem = this.addItem.bind(this);
    this.deleteItem = this.deleteItem.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogout() {
      logout().then(function () {
          localStorage.removeItem(appTokenKey);
          this.props.history.push("/login");
          console.log("user signed out from firebase");
      }.bind(this));
  }

  addItem(e) {
    if(this._inputElement.value !== ""){
      let newItem = {
        text: this._inputElement.value,
        key: Date.now()
      };

      this.setState((prevState) => {
        return{
          items: prevState.items.concat(newItem)
        };
      });
    }
    this._inputElement.value = "";

    console.log(this.state.items);

    e.preventDefault();
  }

  deleteItem(key){
    const filteredItems = this.state.items.filter(function(item) {
      return(item.key !== key)
    });
    this.setState({
      items: filteredItems
    });
  }

  render() {
    const { isLoading } = this.state.loading
    return (
      <div className="todoListMain">
        <Loader />
        <h2>My Todo</h2>
        <div className="Button">
                <RaisedButton
                    backgroundColor="#a4c639"
                    labelColor="#ffffff"
                    label="Sign Out"
                    onTouchTap={this.handleLogout}
                />
              </div>
       <br/>
       <hr/>
       <br/>
        <div className="header">
          <form className="todo_submission">
          What needs to be done?
          <input ref={(a) => this._inputElement = a} className="input-box" value={ this.state.item } onChange={e => this.setState({
            items: e.target.value
          })} >
            </input>
            <button type="submit" onClick={this.addItem}>Add</button>
          </form>
        </div>
        <TodoItems entries={this.state.items} adds={this.addItem} delete={this.deleteItem} />
      </div>
    );
  }
}

export default TodoList;
