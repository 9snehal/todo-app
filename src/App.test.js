import React from 'react';
import ReactDOM from 'react-dom';
import TodoList from './TodoList';
import TodoItems from './TodoItems';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TodoList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly text component', () => {
  const InputTextComponent = renderer.create(<TodoList />).toJSON();
  expect(InputTextComponent).toMatchSnapshot();
});

it('TodoList renders button', () => {
  const add = shallow(<TodoList />);
  const button = add.find('button');
  expect(button).toBeDefined();
});

it('TodoList renders TodoItems Component', () => {
  const app = shallow(<TodoList />);
  expect(app.find('TodoItems').length).toEqual(1);
});

it.skip('Button click calls addItem', () => {
  const add = shallow(<TodoList />);
  const button = add.find('button');
  const input = add.find('input');
  input.simulate('change', { target: { value: 'Read Book' } });
  button.simulate('click');
  expect(addItem).toBeCalledWith('Read Book');
});

it.skip('addItem updates List', () => {
  const app = shallow(<TodoList />);
  const add = app.find('TodoItems');
  add.props.addItem('Read Books');
  app.update();
  const sapp = shallow(<TodoItems />);
  const listData = sapp.props.entries;
  expect(listData.length).toEqual(1);
  expect(listData[0]).toEqual('Read Books');
});

it('should render correctly in "debug" mode', () => {
  const component = shallow(<TodoList debug />);
  expect(component).toMatchSnapshot();
});
