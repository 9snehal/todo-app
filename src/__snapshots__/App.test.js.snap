// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly text component 1`] = `
<div
  className="todoListMain"
>
  <div
    align="center"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
        <circle cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(45 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.125s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(90 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.25s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(135 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.375s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(180 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.5s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(225 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.625s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(270 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.75s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(315 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.875s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
        <circle transform=\\"rotate(180 16 16)\\" cx=\\"16\\" cy=\\"3\\" r=\\"0\\">
          <animate attributeName=\\"r\\" values=\\"0;3;0;0\\" dur=\\"1s\\" repeatCount=\\"indefinite\\" begin=\\"0.5s\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8\\" calcMode=\\"spline\\" />
        </circle>
      </svg>
      ",
      }
    }
    style={
      Object {
        "fill": "blue",
        "height": "20%",
        "width": "20%",
      }
    }
  />
  <h2>
    My Todo
  </h2>
  <br />
  <hr />
  <br />
  <div
    className="header"
  >
    <form
      className="todo_submission"
    >
      What needs to be done?
      <input
        className="input-box"
        onChange={[Function]}
        value={undefined}
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todoList"
  />
</div>
`;

exports[`should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    debug={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loader />,
        <h2>
          My Todo
</h2>,
        <br />,
        <hr />,
        <br />,
        <div
          className="header"
>
          <form
                    className="todo_submission"
          >
                    What needs to be done?
                    <input
                              className="input-box"
                              onChange={[Function]}
                              value={undefined}
                    />
                    <button
                              onClick={[Function]}
                              type="submit"
                    >
                              Add
                    </button>
          </form>
</div>,
        <TodoItems
          adds={[Function]}
          delete={[Function]}
          entries={Array []}
/>,
      ],
      "className": "todoListMain",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Todo",
        },
        "ref": null,
        "rendered": "My Todo",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="todo_submission"
>
            What needs to be done?
            <input
                        className="input-box"
                        onChange={[Function]}
                        value={undefined}
            />
            <button
                        onClick={[Function]}
                        type="submit"
            >
                        Add
            </button>
</form>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "What needs to be done?",
              <input
                className="input-box"
                onChange={[Function]}
                value={undefined}
/>,
              <button
                onClick={[Function]}
                type="submit"
>
                Add
</button>,
            ],
            "className": "todo_submission",
          },
          "ref": null,
          "rendered": Array [
            "What needs to be done?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-box",
                "onChange": [Function],
                "value": undefined,
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "adds": [Function],
          "delete": [Function],
          "entries": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loader />,
          <h2>
            My Todo
</h2>,
          <br />,
          <hr />,
          <br />,
          <div
            className="header"
>
            <form
                        className="todo_submission"
            >
                        What needs to be done?
                        <input
                                    className="input-box"
                                    onChange={[Function]}
                                    value={undefined}
                        />
                        <button
                                    onClick={[Function]}
                                    type="submit"
                        >
                                    Add
                        </button>
            </form>
</div>,
          <TodoItems
            adds={[Function]}
            delete={[Function]}
            entries={Array []}
/>,
        ],
        "className": "todoListMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Todo",
          },
          "ref": null,
          "rendered": "My Todo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="todo_submission"
>
              What needs to be done?
              <input
                            className="input-box"
                            onChange={[Function]}
                            value={undefined}
              />
              <button
                            onClick={[Function]}
                            type="submit"
              >
                            Add
              </button>
</form>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "What needs to be done?",
                <input
                  className="input-box"
                  onChange={[Function]}
                  value={undefined}
/>,
                <button
                  onClick={[Function]}
                  type="submit"
>
                  Add
</button>,
              ],
              "className": "todo_submission",
            },
            "ref": null,
            "rendered": Array [
              "What needs to be done?",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-box",
                  "onChange": [Function],
                  "value": undefined,
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Add",
                "type": "button",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "adds": [Function],
            "delete": [Function],
            "entries": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
